File: Msys_Programs/MSys_Projects/PythonProg/Python_DSA/Sorting_Algorithm/Insertion_Sort.py
def insertionsort(a):
    n = len(a)
    for i in range(1,n):
        cvalue = a[i]
        position = i
        while position > 0 and a[position-1] > cvalue:
            a[position] = a[position-1]
            position = position -1
        a[position] = cvalue

a = [5.0,6,8,2,5,4,9]
print("Orignal list\n", a)
insertionsort(a)
print("After insertion sorting\n",a)

File: Msys_Programs/MSys_Projects/GoProg/DataStructure/SortingAlgorithm/Insertion_Sort.go
package main

import "fmt"

func insertionsort(a []int) {
	for i := 1; i < len(a); i++ {
		cvalue := a[i]
		position := i
		for position > 0 && a[position-1] > cvalue {
			a[position] = a[position-1]
			position = position - 1
		}
		a[position] = cvalue
	}
}

func main() {
	a := []int{5, 6, 2, 8, 4, 3}
	fmt.Println("Before sort", a)
	insertionsort(a)
	fmt.Println("After sort", a)

}
